CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+cmake_minimum_required(VERSION 3.5)
+
+set(CMAKE_CXX_STANDARD 17)
+
+project(
+  pyvinecopulib
+  VERSION ${SKBUILD_PROJECT_VERSION}
+  LANGUAGES CXX
+)
+
+if (NOT SKBUILD)
+  message(WARNING "\
+  This CMake file is meant to be executed using 'scikit-build'. Running
+  it directly will almost certainly not produce the desired result. If
+  you are a user trying to install this package, please use the command
+  below, which will install all necessary build dependencies, compile
+  the package in an isolated environment, and then install it.
+  =====================================================================
+   $ pip install .
+  =====================================================================
+  If you are a software developer, and this is your own package, then
+  it is usually much more efficient to install the build dependencies
+  in your environment once and use the following command that avoids
+  a costly creation of a new virtual environment at every compilation:
+  =====================================================================
+   $ pip install nanobind scikit-build-core[pyproject]
+   $ pip install --no-build-isolation -ve .
+  =====================================================================
+  You may optionally add -Ceditable.rebuild=true to auto-rebuild when
+  the package is imported. Otherwise, you need to re-run the above
+  after editing C++ files.")
+endif()
+
+include(lib/vinecopulib/cmake/options.cmake               REQUIRED)
+
+include(lib/vinecopulib/cmake/compilerDefOpt.cmake        REQUIRED)
+
+# wdm_INCLUDE_DIRS is lib/wdm/include
+set(wdm_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/lib/wdm/include)
+set(vinecopulib_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/lib/vinecopulib/include)
+set(pyvinecopulib_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/src/include)
+
+# BUILD_TESTING is turned off
+set(BUILD_TESTING OFF)
+
+# Look for Eigen3 and Boost
+if(DEFINED ENV{EIGEN3_INCLUDE_DIR})
+  set(EIGEN3_INCLUDE_DIR $ENV{EIGEN3_INCLUDE_DIR})
+endif()
+
+if(DEFINED ENV{Boost_INCLUDE_DIR})
+  set(Boost_INCLUDE_DIRS $ENV{Boost_INCLUDE_DIR})
```

Comment:
* is the distinction DIRS/DIR intentional here?
* for Eigen: what does that even do, if the env var is already defined?

Reply:
* Boost and Eigen do not use the same convention.
* It looks for the env var and makes it available in cmake without having to pass it via -DBoost_INCLUDE_DIRS=${Boost_INCLUDE_DIR} when calling cmake